tls/extensions/s2n_client_supported_groups.c

```diff
@@ -127,6 +127,7 @@ static int s2n_choose_supported_group(struct s2n_connection *conn) {
     const struct s2n_kem_preferences *kem_pref = NULL;
     POSIX_GUARD(s2n_connection_get_kem_preferences(conn, &kem_pref));
     POSIX_ENSURE_REF(kem_pref);
+    POSIX_ENSURE(kem_pref->tls13_kem_group_count <= S2N_SUPPORTED_KEM_GROUPS_COUNT, S2N_ERR_ARRAY_INDEX_OOB);
```

Comment:
I like this assertion, but I'm not sure this is the correct place for it. Seems like it should go higher up in the initialization stack, maybe in [`validate_kem_preferences`](https://github.com/aws/s2n-tls/blob/aab3190ae86fdefe4b46c9689139bb7c57f94a38/tls/s2n_security_policies.c#L820).